### PR TITLE
Implement CoreDataTypes resolving

### LINKED ISSUE #279: In RDF there are core types (i.e. types that might be omited when specifying literal):
In RDF there are core types (i.e. types that might be omited when specifying literal):
image

The strategy should be as follows:
1) core type should have term definition in context
`@id`
2) non-core types should have term definition in context
`@id`
`@type`

_Originally posted by @blcham in https://github.com/kbss-cvut/ava-analytics-ui/issues/83#issuecomment-2360988893_

### PR DESCRIPTION
**Pull request by palagdan**
Resolve #279

### CODE REVIEW

**Comment by palagdan** [comment]
@blcham 

t’s strange that boolean is resolved like core types, but float is not.

This is the resulting JSON-LD when I tested most of the types. It should work as expected:

```java

    "@id": "_:b1",
    "@type": "ValidationConstraintError",
    "columns": [
        "person",
        "lastName",
        "firstName",
        "integerTest",
        "intTest",
        "longTest",
        "dateTest",
        "byteTest",
        "doubleTest",
        "floatTest",
        "shortTest"
    ],
    "constraintFailureEvidences": [
        {
            "@id": "_:b0",
            "byteTest": "123",
            "dateTest": "2023-12-8",
            "doubleTest": 12.333,
            "firstName": "Peter",
            "floatTest": "12.333",
            "intTest": "133",
            "integerTest": 12,
            "lastName": "Hnizdo",
            "longTest": "145",
            "person": "http://onto.fel.cvut.cz/ontologies/s-pipes/examples/constraint-validation/people/person-2",
            "shortTest": "1234"
        },
        {
            "@id": "_:b2",
            "byteTest": "123",
            "dateTest": "2023-12-8",
            "doubleTest": 12.333,
            "firstName": "Pavel",
            "floatTest": "12.333",
            "intTest": "133",
            "integerTest": 12,
            "lastName": "Hnizdo",
            "longTest": "145",
            "person": "http://onto.fel.cvut.cz/ontologies/s-pipes/examples/constraint-validation/people/person-1",
            "shortTest": "1234"
        }
    ],
    "constraintQuery": "# More than one person matches input parameters\n        PREFIX foaf: <http://xmlns.com/foaf/0.1/>\n\n        SELECT ?person (?lastNameOuter as ?lastName) ?firstName ?integerTest ?intTest ?longTest ?dateTest ?byteTest ?doubleTest ?floatTest ?shortTest\n        WHERE {\n          {\n            SELECT (count(distinct ?p) as ?pCount)\n            WHERE {\n               ?p a foaf:Person;\n                  foaf:firstName ?firstName;\n                  foaf:lastName ?lastName;\n              .\n            }\n          }\n\n          FILTER(?pCount > 1)\n\n          ?person a foaf:Person;\n             foaf:lastName ?lastNameOuter;\n             foaf:firstName ?firstName;\n            BIND(\"12\"^^xsd:integer AS ?integerTest)\n            BIND(\"133\"^^xsd:int as ?intTest)\n            BIND(\"145\"^^xsd:long as ?longTest)\n            BIND(\"2023-12-8\"^^xsd:date as ?dateTest)\n            BIND(\"123\"^^xsd:byte as ?byteTest)\n            BIND(\"12.333\"^^xsd:double as ?doubleTest)\n            BIND(\"12.333\"^^xsd:float as ?floatTest)\n            BIND(\"1234\"^^xsd:short as ?shortTest)\n\n          .\n      }",
    "message": "More than one person matches input parameters",
    "module": "http://onto.fel.cvut.cz/ontologies/s-pipes/constraint-validation/construct-matched-person",
    "@context": {
        "ValidationConstraintError": "http://onto.fel.cvut.cz/ontologies/s-pipes/ValidationConstraintError",
        "module": "http://onto.fel.cvut.cz/ontologies/s-pipes/module",
        "message": "http://onto.fel.cvut.cz/ontologies/s-pipes/message",
        "constraintFailureEvidences": {
            "@id": "http://onto.fel.cvut.cz/ontologies/s-pipes/constraintFailureEvidences",
            "@container": "@list"
        },
        "columns": {
            "@id": "http://onto.fel.cvut.cz/ontologies/s-pipes/columns",
            "@container": "@list"
        },
        "person": {
            "@id": "http://onto.fel.cvut.cz/ontologies/s-pipes/person",
            "@type": "@id"
        },
        "lastName": {
            "@id": "http://onto.fel.cvut.cz/ontologies/s-pipes/lastName"
        },
        "firstName": {
            "@id": "http://onto.fel.cvut.cz/ontologies/s-pipes/firstName"
        },
        "integerTest": {
            "@id": "http://onto.fel.cvut.cz/ontologies/s-pipes/integerTest"
        },
        "intTest": {
            "@id": "http://onto.fel.cvut.cz/ontologies/s-pipes/intTest",
            "@type": "http://www.w3.org/2001/XMLSchema#int"
        },
        "longTest": {
            "@id": "http://onto.fel.cvut.cz/ontologies/s-pipes/longTest",
            "@type": "http://www.w3.org/2001/XMLSchema#long"
        },
        "dateTest": {
            "@id": "http://onto.fel.cvut.cz/ontologies/s-pipes/dateTest",
            "@type": "http://www.w3.org/2001/XMLSchema#date"
        },
        "byteTest": {
            "@id": "http://onto.fel.cvut.cz/ontologies/s-pipes/byteTest",
            "@type": "http://www.w3.org/2001/XMLSchema#byte"
        },
        "doubleTest": {
            "@id": "http://onto.fel.cvut.cz/ontologies/s-pipes/doubleTest"
        },
        "floatTest": {
            "@id": "http://onto.fel.cvut.cz/ontologies/s-pipes/floatTest",
            "@type": "http://www.w3.org/2001/XMLSchema#float"
        },
        "shortTest": {
            "@id": "http://onto.fel.cvut.cz/ontologies/s-pipes/shortTest",
            "@type": "http://www.w3.org/2001/XMLSchema#short"
        },
        "constraintQuery": "http://onto.fel.cvut.cz/ontologies/s-pipes/constraintQuery"
    }
}
```

**Comment by blcham** [comment]
@palagdan As suggested in https://github.com/kbss-cvut/s-pipes/issues/279#issue-2538127696, it is crucial to test:
[ ] xsd:string
[ ] xsd:boolean
[ ] xsd:double
[ ] xsd:integer

**Comment by palagdan** [comment]
@blcham 

It seems that the boolean is also working as expected:

```json
{
    "@id": "_:b1",
    "@type": "ValidationConstraintError",
    "columns": [
        "person",
        "lastName",
        "firstName",
        "integerTest",
        "intTest",
        "longTest",
        "dateTest",
        "byteTest",
        "doubleTest",
        "floatTest",
        "shortTest",
        "booleanTest"
    ],
    "constraintFailureEvidences": [
        {
            "@id": "_:b2",
            "booleanTest": false,
            "byteTest": "123",
            "dateTest": "2023-12-8",
            "doubleTest": 12.333,
            "firstName": "Peter",
            "floatTest": "12.333",
            "intTest": "133",
            "integerTest": 12,
            "lastName": "Hnizdo",
            "longTest": "145",
            "person": "http://onto.fel.cvut.cz/ontologies/s-pipes/examples/constraint-validation/people/person-2",
            "shortTest": "1234"
        },
        {
            "@id": "_:b0",
            "booleanTest": false,
            "byteTest": "123",
            "dateTest": "2023-12-8",
            "doubleTest": 12.333,
            "firstName": "Pavel",
            "floatTest": "12.333",
            "intTest": "133",
            "integerTest": 12,
            "lastName": "Hnizdo",
            "longTest": "145",
            "person": "http://onto.fel.cvut.cz/ontologies/s-pipes/examples/constraint-validation/people/person-1",
            "shortTest": "1234"
        }
    ],
    "constraintQuery": "# More than one person matches input parameters\n        PREFIX foaf: <http://xmlns.com/foaf/0.1/>\n\n        SELECT ?person (?lastNameOuter as ?lastName) ?firstName ?integerTest ?intTest ?longTest ?dateTest ?byteTest ?doubleTest ?floatTest ?shortTest ?booleanTest\n        WHERE {\n          {\n            SELECT (count(distinct ?p) as ?pCount)\n            WHERE {\n               ?p a foaf:Person;\n                  foaf:firstName ?firstName;\n                  foaf:lastName ?lastName;\n              .\n            }\n          }\n\n          FILTER(?pCount > 1)\n\n          ?person a foaf:Person;\n             foaf:lastName ?lastNameOuter;\n             foaf:firstName ?firstName;\n            BIND(\"12\"^^xsd:integer AS ?integerTest)\n            BIND(\"133\"^^xsd:int as ?intTest)\n            BIND(\"145\"^^xsd:long as ?longTest)\n            BIND(\"2023-12-8\"^^xsd:date as ?dateTest)\n            BIND(\"123\"^^xsd:byte as ?byteTest)\n            BIND(\"12.333\"^^xsd:double as ?doubleTest)\n            BIND(\"12.333\"^^xsd:float as ?floatTest)\n            BIND(\"1234\"^^xsd:short as ?shortTest)\n            BIND(\"false\"^^xsd:boolean as ?booleanTest)\n\n          .\n      }",
    "message": "More than one person matches input parameters",
    "module": "http://onto.fel.cvut.cz/ontologies/s-pipes/constraint-validation/construct-matched-person",
    "@context": {
        "ValidationConstraintError": "http://onto.fel.cvut.cz/ontologies/s-pipes/ValidationConstraintError",
        "module": "http://onto.fel.cvut.cz/ontologies/s-pipes/module",
        "message": "http://onto.fel.cvut.cz/ontologies/s-pipes/message",
        "constraintFailureEvidences": {
            "@id": "http://onto.fel.cvut.cz/ontologies/s-pipes/constraintFailureEvidences",
            "@container": "@list"
        },
        "columns": {
            "@id": "http://onto.fel.cvut.cz/ontologies/s-pipes/columns",
            "@container": "@list"
        },
        "person": {
            "@id": "http://onto.fel.cvut.cz/ontologies/s-pipes/person",
            "@type": "@id"
        },
        "lastName": {
            "@id": "http://onto.fel.cvut.cz/ontologies/s-pipes/lastName"
        },
        "firstName": {
            "@id": "http://onto.fel.cvut.cz/ontologies/s-pipes/firstName"
        },
        "integerTest": {
            "@id": "http://onto.fel.cvut.cz/ontologies/s-pipes/integerTest"
        },
        "intTest": {
            "@id": "http://onto.fel.cvut.cz/ontologies/s-pipes/intTest",
            "@type": "http://www.w3.org/2001/XMLSchema#int"
        },
        "longTest": {
            "@id": "http://onto.fel.cvut.cz/ontologies/s-pipes/longTest",
            "@type": "http://www.w3.org/2001/XMLSchema#long"
        },
        "dateTest": {
            "@id": "http://onto.fel.cvut.cz/ontologies/s-pipes/dateTest",
            "@type": "http://www.w3.org/2001/XMLSchema#date"
        },
        "byteTest": {
            "@id": "http://onto.fel.cvut.cz/ontologies/s-pipes/byteTest",
            "@type": "http://www.w3.org/2001/XMLSchema#byte"
        },
        "doubleTest": {
            "@id": "http://onto.fel.cvut.cz/ontologies/s-pipes/doubleTest"
        },
        "floatTest": {
            "@id": "http://onto.fel.cvut.cz/ontologies/s-pipes/floatTest",
            "@type": "http://www.w3.org/2001/XMLSchema#float"
        },
        "shortTest": {
            "@id": "http://onto.fel.cvut.cz/ontologies/s-pipes/shortTest",
            "@type": "http://www.w3.org/2001/XMLSchema#short"
        },
        "booleanTest": {
            "@id": "http://onto.fel.cvut.cz/ontologies/s-pipes/booleanTest"
        },
        "constraintQuery": "http://onto.fel.cvut.cz/ontologies/s-pipes/constraintQuery"
    }
}
```

**Comment by palagdan** [comment]
Decimal is not working as expected. The @type should also be marked. That is why I tested all possible types:
```json
{
    "@id": "_:b2",
    "@type": "ValidationConstraintError",
    "columns": [
        "person",
        "lastName",
        "firstName",
        "integerTest",
        "intTest",
        "longTest",
        "dateTest",
        "byteTest",
        "doubleTest",
        "floatTest",
        "shortTest",
        "booleanTest",
        "decimalTest",
        "stringTest",
        "gYearTest",
        "gMonthTest",
        "gDayTest",
        "gYearMonthTest",
        "gMonthDayTest",
        "timeTest",
        "dateTimeTest",
        "dateTimeStampTest",
        "durationTest",
        "yearMonthDurationTest",
        "dayTimeDurationTest",
        "unsignedByteTest",
        "unsignedShortTest",
        "unsignedIntTest",
        "unsignedLongTest",
        "positiveIntegerTest",
        "nonNegativeIntegerTest",
        "negativeIntegerTest",
        "nonPositiveIntegerTest",
        "hexBinaryTest",
        "base64BinaryTest",
        "anyURITest",
        "languageTest",
        "normalizedStringTest",
        "tokenTest",
        "NMTOKENTest",
        "NameTest",
        "NCNameTest"
    ],
    "constraintFailureEvidences": [
        {
            "@id": "_:b1",
            "NCNameTest": "NCNameValue",
            "NMTOKENTest": "NMTOKENValue",
            "NameTest": "NameValue",
            "anyURITest": "http://example.org",
            "base64BinaryTest": "SGVsbG8gd29ybGQ=",
            "booleanTest": false,
            "byteTest": "123",
            "dateTest": "2023-12-8",
            "dateTimeStampTest": "2023-12-08T12:30:45Z",
            "dateTimeTest": "2023-12-08T12:30:45",
            "dayTimeDurationTest": "P4DT5H6M",
            "decimalTest": "12678967.543233",
            "doubleTest": 12.333,
            "durationTest": "P1Y2M3DT4H5M6S",
            "firstName": "Peter",
            "floatTest": "12.333",
            "gDayTest": "8",
            "gMonthDayTest": "12-8",
            "gMonthTest": "12",
            "gYearMonthTest": "2023-12",
            "gYearTest": "2023",
            "hexBinaryTest": "48656c6c6f",
            "intTest": "133",
            "integerTest": 12,
            "languageTest": "en-US",
            "lastName": "Hnizdo",
            "longTest": "145",
            "negativeIntegerTest": "-1",
            "nonNegativeIntegerTest": "0",
            "nonPositiveIntegerTest": "0",
            "normalizedStringTest": "HelloWorld",
            "person": "http://onto.fel.cvut.cz/ontologies/s-pipes/examples/constraint-validation/people/person-2",
            "positiveIntegerTest": "1",
            "shortTest": "1234",
            "stringTest": "Hello World",
            "timeTest": "12:30:45",
            "tokenTest": "TokenValue",
            "unsignedByteTest": "255",
            "unsignedIntTest": "4294967295",
            "unsignedLongTest": "18446744073709551615",
            "unsignedShortTest": "65535",
            "yearMonthDurationTest": "P2Y3M"
        },
        {
            "@id": "_:b0",
            "NCNameTest": "NCNameValue",
            "NMTOKENTest": "NMTOKENValue",
            "NameTest": "NameValue",
            "anyURITest": "http://example.org",
            "base64BinaryTest": "SGVsbG8gd29ybGQ=",
            "booleanTest": false,
            "byteTest": "123",
            "dateTest": "2023-12-8",
            "dateTimeStampTest": "2023-12-08T12:30:45Z",
            "dateTimeTest": "2023-12-08T12:30:45",
            "dayTimeDurationTest": "P4DT5H6M",
            "decimalTest": "12678967.543233",
            "doubleTest": 12.333,
            "durationTest": "P1Y2M3DT4H5M6S",
            "firstName": "Pavel",
            "floatTest": "12.333",
            "gDayTest": "8",
            "gMonthDayTest": "12-8",
            "gMonthTest": "12",
            "gYearMonthTest": "2023-12",
            "gYearTest": "2023",
            "hexBinaryTest": "48656c6c6f",
            "intTest": "133",
            "integerTest": 12,
            "languageTest": "en-US",
            "lastName": "Hnizdo",
            "longTest": "145",
            "negativeIntegerTest": "-1",
            "nonNegativeIntegerTest": "0",
            "nonPositiveIntegerTest": "0",
            "normalizedStringTest": "HelloWorld",
            "person": "http://onto.fel.cvut.cz/ontologies/s-pipes/examples/constraint-validation/people/person-1",
            "positiveIntegerTest": "1",
            "shortTest": "1234",
            "stringTest": "Hello World",
            "timeTest": "12:30:45",
            "tokenTest": "TokenValue",
            "unsignedByteTest": "255",
            "unsignedIntTest": "4294967295",
            "unsignedLongTest": "18446744073709551615",
            "unsignedShortTest": "65535",
            "yearMonthDurationTest": "P2Y3M"
        }
    ],
    "constraintQuery": "# More than one person matches input parameters\n        PREFIX foaf: <http://xmlns.com/foaf/0.1/>\n\n        SELECT\n          ?person\n          (?lastNameOuter as ?lastName)\n          ?firstName\n          ?integerTest\n          ?intTest\n          ?longTest\n          ?dateTest\n          ?byteTest\n          ?doubleTest\n          ?floatTest\n          ?shortTest\n          ?booleanTest\n          ?decimalTest\n          ?stringTest\n          ?gYearTest\n          ?gMonthTest\n          ?gDayTest\n          ?gYearMonthTest\n          ?gMonthDayTest\n          ?timeTest\n          ?dateTimeTest\n          ?dateTimeStampTest\n          ?durationTest\n          ?yearMonthDurationTest\n          ?dayTimeDurationTest\n          ?unsignedByteTest\n          ?unsignedShortTest\n          ?unsignedIntTest\n          ?unsignedLongTest\n          ?positiveIntegerTest\n          ?nonNegativeIntegerTest\n          ?negativeIntegerTest\n          ?nonPositiveIntegerTest\n          ?hexBinaryTest\n          ?base64BinaryTest\n          ?anyURITest\n          ?languageTest\n          ?normalizedStringTest\n          ?tokenTest\n          ?NMTOKENTest\n          ?NameTest\n          ?NCNameTest\n        WHERE {\n          {\n            SELECT (count(distinct ?p) as ?pCount)\n            WHERE {\n              ?p a foaf:Person;\n                 foaf:firstName ?firstName;\n                 foaf:lastName ?lastName;\n            }\n          }\n          FILTER(?pCount > 1)\n\n          ?person a foaf:Person;\n             foaf:lastName ?lastNameOuter;\n             foaf:firstName ?firstName;\n\n          # Existing BINDs\n          BIND(\"12\"^^xsd:integer AS ?integerTest)\n          BIND(\"133\"^^xsd:int AS ?intTest)\n          BIND(\"145\"^^xsd:long AS ?longTest)\n          BIND(\"2023-12-8\"^^xsd:date AS ?dateTest)\n          BIND(\"123\"^^xsd:byte AS ?byteTest)\n          BIND(\"12.333\"^^xsd:double AS ?doubleTest)\n          BIND(\"12.333\"^^xsd:float AS ?floatTest)\n          BIND(\"1234\"^^xsd:short AS ?shortTest)\n          BIND(\"false\"^^xsd:boolean AS ?booleanTest)\n          BIND(\"12678967.543233\"^^xsd:decimal AS ?decimalTest)\n\n          # New BINDs for other types\n          BIND(\"Hello World\"^^xsd:string AS ?stringTest)\n          BIND(\"2023\"^^xsd:gYear AS ?gYearTest)\n          BIND(\"12\"^^xsd:gMonth AS ?gMonthTest)\n          BIND(\"8\"^^xsd:gDay AS ?gDayTest)\n          BIND(\"2023-12\"^^xsd:gYearMonth AS ?gYearMonthTest)\n          BIND(\"12-8\"^^xsd:gMonthDay AS ?gMonthDayTest)\n          BIND(\"12:30:45\"^^xsd:time AS ?timeTest)\n          BIND(\"2023-12-08T12:30:45\"^^xsd:dateTime AS ?dateTimeTest)\n          BIND(\"2023-12-08T12:30:45Z\"^^xsd:dateTimeStamp AS ?dateTimeStampTest)\n          BIND(\"P1Y2M3DT4H5M6S\"^^xsd:duration AS ?durationTest)\n          BIND(\"P2Y3M\"^^xsd:yearMonthDuration AS ?yearMonthDurationTest)\n          BIND(\"P4DT5H6M\"^^xsd:dayTimeDuration AS ?dayTimeDurationTest)\n\n          BIND(\"255\"^^xsd:unsignedByte AS ?unsignedByteTest)\n          BIND(\"65535\"^^xsd:unsignedShort AS ?unsignedShortTest)\n          BIND(\"4294967295\"^^xsd:unsignedInt AS ?unsignedIntTest)\n          BIND(\"18446744073709551615\"^^xsd:unsignedLong AS ?unsignedLongTest)\n          BIND(\"1\"^^xsd:positiveInteger AS ?positiveIntegerTest)\n          BIND(\"0\"^^xsd:nonNegativeInteger AS ?nonNegativeIntegerTest)\n          BIND(\"-1\"^^xsd:negativeInteger AS ?negativeIntegerTest)\n          BIND(\"0\"^^xsd:nonPositiveInteger AS ?nonPositiveIntegerTest)\n\n          BIND(\"48656c6c6f\"^^xsd:hexBinary AS ?hexBinaryTest)\n          BIND(\"SGVsbG8gd29ybGQ=\"^^xsd:base64Binary AS ?base64BinaryTest)\n\n          BIND(\"http://example.org\"^^xsd:anyURI AS ?anyURITest)\n          BIND(\"en-US\"^^xsd:language AS ?languageTest)\n          BIND(\"HelloWorld\"^^xsd:normalizedString AS ?normalizedStringTest)\n          BIND(\"TokenValue\"^^xsd:token AS ?tokenTest)\n          BIND(\"NMTOKENValue\"^^xsd:NMTOKEN AS ?NMTOKENTest)\n          BIND(\"NameValue\"^^xsd:Name AS ?NameTest)\n          BIND(\"NCNameValue\"^^xsd:NCName AS ?NCNameTest)\n        \n          .\n      }",
    "message": "More than one person matches input parameters",
    "module": "http://onto.fel.cvut.cz/ontologies/s-pipes/constraint-validation/construct-matched-person",
    "@context": {
        "ValidationConstraintError": "http://onto.fel.cvut.cz/ontologies/s-pipes/ValidationConstraintError",
        "module": "http://onto.fel.cvut.cz/ontologies/s-pipes/module",
        "message": "http://onto.fel.cvut.cz/ontologies/s-pipes/message",
        "constraintFailureEvidences": {
            "@id": "http://onto.fel.cvut.cz/ontologies/s-pipes/constraintFailureEvidences",
            "@container": "@list"
        },
        "columns": {
            "@id": "http://onto.fel.cvut.cz/ontologies/s-pipes/columns",
            "@container": "@list"
        },
        "person": {
            "@id": "http://onto.fel.cvut.cz/ontologies/s-pipes/person",
            "@type": "@id"
        },
        "lastName": {
            "@id": "http://onto.fel.cvut.cz/ontologies/s-pipes/lastName"
        },
        "firstName": {
            "@id": "http://onto.fel.cvut.cz/ontologies/s-pipes/firstName"
        },
        "integerTest": {
            "@id": "http://onto.fel.cvut.cz/ontologies/s-pipes/integerTest"
        },
        "intTest": {
            "@id": "http://onto.fel.cvut.cz/ontologies/s-pipes/intTest",
            "@type": "http://www.w3.org/2001/XMLSchema#int"
        },
        "longTest": {
            "@id": "http://onto.fel.cvut.cz/ontologies/s-pipes/longTest",
            "@type": "http://www.w3.org/2001/XMLSchema#long"
        },
        "dateTest": {
            "@id": "http://onto.fel.cvut.cz/ontologies/s-pipes/dateTest",
            "@type": "http://www.w3.org/2001/XMLSchema#date"
        },
        "byteTest": {
            "@id": "http://onto.fel.cvut.cz/ontologies/s-pipes/byteTest",
            "@type": "http://www.w3.org/2001/XMLSchema#byte"
        },
        "doubleTest": {
            "@id": "http://onto.fel.cvut.cz/ontologies/s-pipes/doubleTest"
        },
        "floatTest": {
            "@id": "http://onto.fel.cvut.cz/ontologies/s-pipes/floatTest",
            "@type": "http://www.w3.org/2001/XMLSchema#float"
        },
        "shortTest": {
            "@id": "http://onto.fel.cvut.cz/ontologies/s-pipes/shortTest",
            "@type": "http://www.w3.org/2001/XMLSchema#short"
        },
        "booleanTest": {
            "@id": "http://onto.fel.cvut.cz/ontologies/s-pipes/booleanTest"
        },
        "decimalTest": {
            "@id": "http://onto.fel.cvut.cz/ontologies/s-pipes/decimalTest",
            "@type": "http://www.w3.org/2001/XMLSchema#decimal"
        },
        "stringTest": {
            "@id": "http://onto.fel.cvut.cz/ontologies/s-pipes/stringTest"
        },
        "gYearTest": {
            "@id": "http://onto.fel.cvut.cz/ontologies/s-pipes/gYearTest",
            "@type": "http://www.w3.org/2001/XMLSchema#gYear"
        },
        "gMonthTest": {
            "@id": "http://onto.fel.cvut.cz/ontologies/s-pipes/gMonthTest",
            "@type": "http://www.w3.org/2001/XMLSchema#gMonth"
        },
        "gDayTest": {
            "@id": "http://onto.fel.cvut.cz/ontologies/s-pipes/gDayTest",
            "@type": "http://www.w3.org/2001/XMLSchema#gDay"
        },
        "gYearMonthTest": {
            "@id": "http://onto.fel.cvut.cz/ontologies/s-pipes/gYearMonthTest",
            "@type": "http://www.w3.org/2001/XMLSchema#gYearMonth"
        },
        "gMonthDayTest": {
            "@id": "http://onto.fel.cvut.cz/ontologies/s-pipes/gMonthDayTest",
            "@type": "http://www.w3.org/2001/XMLSchema#gMonthDay"
        },
        "timeTest": {
            "@id": "http://onto.fel.cvut.cz/ontologies/s-pipes/timeTest",
            "@type": "http://www.w3.org/2001/XMLSchema#time"
        },
        "dateTimeTest": {
            "@id": "http://onto.fel.cvut.cz/ontologies/s-pipes/dateTimeTest",
            "@type": "http://www.w3.org/2001/XMLSchema#dateTime"
        },
        "dateTimeStampTest": {
            "@id": "http://onto.fel.cvut.cz/ontologies/s-pipes/dateTimeStampTest",
            "@type": "http://www.w3.org/2001/XMLSchema#dateTimeStamp"
        },
        "durationTest": {
            "@id": "http://onto.fel.cvut.cz/ontologies/s-pipes/durationTest",
            "@type": "http://www.w3.org/2001/XMLSchema#duration"
        },
        "yearMonthDurationTest": {
            "@id": "http://onto.fel.cvut.cz/ontologies/s-pipes/yearMonthDurationTest",
            "@type": "http://www.w3.org/2001/XMLSchema#yearMonthDuration"
        },
        "dayTimeDurationTest": {
            "@id": "http://onto.fel.cvut.cz/ontologies/s-pipes/dayTimeDurationTest",
            "@type": "http://www.w3.org/2001/XMLSchema#dayTimeDuration"
        },
        "unsignedByteTest": {
            "@id": "http://onto.fel.cvut.cz/ontologies/s-pipes/unsignedByteTest",
            "@type": "http://www.w3.org/2001/XMLSchema#unsignedByte"
        },
        "unsignedShortTest": {
            "@id": "http://onto.fel.cvut.cz/ontologies/s-pipes/unsignedShortTest",
            "@type": "http://www.w3.org/2001/XMLSchema#unsignedShort"
        },
        "unsignedIntTest": {
            "@id": "http://onto.fel.cvut.cz/ontologies/s-pipes/unsignedIntTest",
            "@type": "http://www.w3.org/2001/XMLSchema#unsignedInt"
        },
        "unsignedLongTest": {
            "@id": "http://onto.fel.cvut.cz/ontologies/s-pipes/unsignedLongTest",
            "@type": "http://www.w3.org/2001/XMLSchema#unsignedLong"
        },
        "positiveIntegerTest": {
            "@id": "http://onto.fel.cvut.cz/ontologies/s-pipes/positiveIntegerTest",
            "@type": "http://www.w3.org/2001/XMLSchema#positiveInteger"
        },
        "nonNegativeIntegerTest": {
            "@id": "http://onto.fel.cvut.cz/ontologies/s-pipes/nonNegativeIntegerTest",
            "@type": "http://www.w3.org/2001/XMLSchema#nonNegativeInteger"
        },
        "negativeIntegerTest": {
            "@id": "http://onto.fel.cvut.cz/ontologies/s-pipes/negativeIntegerTest",
            "@type": "http://www.w3.org/2001/XMLSchema#negativeInteger"
        },
        "nonPositiveIntegerTest": {
            "@id": "http://onto.fel.cvut.cz/ontologies/s-pipes/nonPositiveIntegerTest",
            "@type": "http://www.w3.org/2001/XMLSchema#nonPositiveInteger"
        },
        "hexBinaryTest": {
            "@id": "http://onto.fel.cvut.cz/ontologies/s-pipes/hexBinaryTest",
            "@type": "http://www.w3.org/2001/XMLSchema#hexBinary"
        },
        "base64BinaryTest": {
            "@id": "http://onto.fel.cvut.cz/ontologies/s-pipes/base64BinaryTest",
            "@type": "http://www.w3.org/2001/XMLSchema#base64Binary"
        },
        "anyURITest": {
            "@id": "http://onto.fel.cvut.cz/ontologies/s-pipes/anyURITest",
            "@type": "http://www.w3.org/2001/XMLSchema#anyURI"
        },
        "languageTest": {
            "@id": "http://onto.fel.cvut.cz/ontologies/s-pipes/languageTest",
            "@type": "http://www.w3.org/2001/XMLSchema#language"
        },
        "normalizedStringTest": {
            "@id": "http://onto.fel.cvut.cz/ontologies/s-pipes/normalizedStringTest",
            "@type": "http://www.w3.org/2001/XMLSchema#normalizedString"
        },
        "tokenTest": {
            "@id": "http://onto.fel.cvut.cz/ontologies/s-pipes/tokenTest",
            "@type": "http://www.w3.org/2001/XMLSchema#token"
        },
        "NMTOKENTest": {
            "@id": "http://onto.fel.cvut.cz/ontologies/s-pipes/NMTOKENTest",
            "@type": "http://www.w3.org/2001/XMLSchema#NMTOKEN"
        },
        "NameTest": {
            "@id": "http://onto.fel.cvut.cz/ontologies/s-pipes/NameTest",
            "@type": "http://www.w3.org/2001/XMLSchema#Name"
        },
        "NCNameTest": {
            "@id": "http://onto.fel.cvut.cz/ontologies/s-pipes/NCNameTest",
            "@type": "http://www.w3.org/2001/XMLSchema#NCName"
        },
        "constraintQuery": "http://onto.fel.cvut.cz/ontologies/s-pipes/constraintQuery"
    }
}
```

**Review by blcham — CHANGES_REQUESTED** [review]
LGTM, can be merged after the comment is resolved.

File: s-pipes-web/src/main/java/cz/cvut/spipes/rest/handler/XsdCoreDataTypes.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
+package cz.cvut.spipes.rest.handler;
+
+import java.util.List;
+
+public class XsdCoreDataTypes {
```
[the remark this answers is not in the document]
Comment (blcham) [reply s-pipes-web/src/main/java/cz/cvut/spipes/rest/handler/XsdCoreDataTypes.java]: I resolved it.

**Comment by blcham** [comment]
@palagdan as u might be offline, i am merging this PR, have a peek if u agree with my last comment

File: s-pipes-web/src/main/java/cz/cvut/spipes/rest/handler/XsdCoreDataTypes.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (palagdan) [reply s-pipes-web/src/main/java/cz/cvut/spipes/rest/handler/XsdCoreDataTypes.java]: @blcham 
I agree with your changes. I think it is a better name for that.

**Pull request merged**